src/test/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceServiceSpec.scala

```diff
@@ -715,6 +715,44 @@ class ResourceServiceSpec extends FlatSpec with Matchers with ScalaFutures with
     managedGroupService.loadManagedGroup(ResourceId(otherAuthDomainGroup), samRequestContext).unsafeRunSync() shouldBe Some(WorkbenchEmail(s"$otherAuthDomainGroup@$emailDomain"))
   }
 
+  it should "delete a child resource that has a parent" in {
+    val parentResource = FullyQualifiedResourceId(defaultResourceType.name, ResourceId("my-resource-parent"))
+    val childResource = FullyQualifiedResourceId(defaultResourceType.name, ResourceId("my-resource-child"))
+    service.createResourceType(defaultResourceType, samRequestContext).unsafeRunSync()
+    runAndWait(service.createResource(defaultResourceType, parentResource.resourceId, dummyUserInfo, samRequestContext))
+    runAndWait(service.createResource(defaultResourceType, childResource.resourceId, dummyUserInfo, samRequestContext))
+    runAndWait(service.setResourceParent(childResource, parentResource, samRequestContext))
+
+    assert(policyDAO.listAccessPolicies(childResource, samRequestContext).unsafeRunSync().nonEmpty)
+
+    runAndWait(service.deleteResource(childResource, samRequestContext))
+
+    assert(policyDAO.listAccessPolicies(childResource, samRequestContext).unsafeRunSync().isEmpty)
+  }
+
+  it should "return 400 for a parent resource that has any children" in {
```

Comment:
also have a test that checks for 400 in the route spec. In the test here I mostly care that it fails and does not delete the resource. In the route spec we care that the response code is 400.